File: atcoder.jp/contests/abc143/tasks/abc143_c/main.cpp
#include <bits/stdc++.h>
using namespace std;

string dedup(const string &s) {
  string result = s;
  result.erase(unique(result.begin(), result.end()), result.end());
  return result;
}

int main() {
  int N;
  string S;
  cin >> N >> S;

  int ans = dedup(S).size();

  cout << ans << endl;
}

File: atcoder.jp/contests/abc143/tasks/abc143_c/main.go
package main

import (
	"fmt"
)

func Dedup(s string) string {
	if len(s) <= 1 {
		return s
	}
	var result []byte
	result = append(result, s[0])
	for i := 1; i < len(s); i++ {
		if s[i-1] != s[i] {
			result = append(result, s[i])
		}
	}
	return string(result)
}

func main() {
	var N int
	fmt.Scanf("%d", &N)
	var S string
	fmt.Scanf("%s", &S)

	ans := len(Dedup(S))

	fmt.Println(ans)
}
